modules/postgresql/src/main/java/org/testcontainers/containers/PostgreSQLContainer.java

```diff
@@ -59,7 +62,24 @@ public String getDriverClassName() {
     @Override
     public String getJdbcUrl() {
         // Disable Postgres driver use of java.util.logging to reduce noise at startup time
-        return "jdbc:postgresql://" + getContainerIpAddress() + ":" + getMappedPort(POSTGRESQL_PORT) + "/" + databaseName + "?loggerLevel=OFF";
+        return format("jdbc:postgresql://%s:%d/%s?loggerLevel=OFF", getContainerIpAddress(), getMappedPort(POSTGRESQL_PORT), databaseName);
+    }
+
+    @Override
+    protected String constructUrlForConnection(String queryString) {
+        String baseUrl = getJdbcUrl();
+
+        if ("".equals(queryString)) {
+            return baseUrl;
+        }
+
+        if (!queryString.startsWith("?")) {
+            throw new IllegalArgumentException("The '?' character must be included");
```

Comment:
Why?

Reply:
Docs: https://github.com/testcontainers/testcontainers-java/blob/f1565f1ba842ccfa9d8ed9ee00276f38b7a2546f/modules/jdbc/src/main/java/org/testcontainers/containers/JdbcDatabaseContainer.java#L218-L219

Reply:
I told you to guard the `substring` to avoid replacing wrong character, but the javadoc is a good argument 👍

Reply:
Actually, the doc can be misleading a bit since these methods are actually called (internally) with an empty string - that would be worth sorting out